principal/event.go

```diff
@@ -230,3 +242,29 @@ func (s *Server) StartEventProcessor(ctx context.Context) error {
 	}()
 	return err
 }
+
+func (s *Server) createNamespaceIfNotExist(ctx context.Context, name string) (bool, error) {
+	if !s.autoNamespaceAllow {
+		return false, nil
+	}
+	if s.autoNamespacePattern != nil {
+		if !s.autoNamespacePattern.MatchString(name) {
+			return false, nil
+		}
+	}
+	_, err := s.kubeClient.CoreV1().Namespaces().Get(ctx, name, v1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return false, err
+		} else {
+			_, err = s.kubeClient.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
+				ObjectMeta: v1.ObjectMeta{
+					Name:   name,
+					Labels: s.autoNamespaceLabels,
+				}},
+				v1.CreateOptions{})
+			return true, err
```

Comment:
Should we be checking if the Namespace is marked for deletion and skip if Yes?

Reply:
Interesting idea. I think if the namespace is marked for deletion, the initial `Get` call would not return an  error (i.e. it would return a resource), and thereby skipping the `Create` logic already.

I think this an edge case we should not tackle at this point. However, I put a TODO comment in the code so that we may revisit this later on.

Are you OK with that?

Reply:
That sounds good to me.